### PR TITLE
Normalize all the arcgis sources for SC

### LINKED ISSUE #756: normalize sc/dab_arcgis
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #749

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch sc/dab_arcgis
    poetry run vaccine-feed-ingest parse sc/dab_arcgis
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize sc/dab_arcgis
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### LINKED ISSUE #757: normalize sc/dhec_arcgis
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #750

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch sc/dhec_arcgis
    poetry run vaccine-feed-ingest parse sc/dhec_arcgis
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize sc/dhec_arcgis
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### LINKED ISSUE #759: normalize sc/nomad_arcgis
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #752

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch sc/nomad_arcgis
    poetry run vaccine-feed-ingest parse sc/nomad_arcgis
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize sc/nomad_arcgis
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by MoralCode**
| Key Details |
|-|
| Resolves #759 Resolves #757 Resolves #756 |
State: sc |
Site: dab_arcgis shec_arcgis and nomad_arcgis |

## Notes
turns out theyre all pretty much the same as the existing arcgis site. i think nomad might have a few more sites (just from looking at the number of lines in the parsed output) but i didnt want to assume there are no differences though so might as well grab them all.

All use the same normalizer except for nomad which has a slight change to include the date it was published (and of course different source names and urls)

### Further improvement suggestions 
Since the fetch stage's `.yml` supports multiple serviceItemID's all of these can easily be merged with the only block on this being some way to associate the correct source names and urls with the right data.

This could be an if-else chain based on the service Item ID, but it seems to make more sense to add a way to add these values to the yml and pass them through since that may be helpful for anyone trying to get from service item id back to the original arcgis map that it came from.

## Data sample

```json
{"id": "sc_nomad_arcgis:1256d5b3c554475099c24b98326e3b36_0_b456ea0b-db46-42f4-9bbe-3df1a38e4fa8", "name": "Carolina Cataract and Vision Center", "address": {"street1": "137 gateway drive", "street2": null, "city": "Ladson", "state": "SC", "zip": "29456"}, "location": {"latitude": 32.997858505756824, "longitude": -80.09152585378803}, "contact": null, "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": null, "active": true, "source": {"source": "sc_nomad_arcgis", "id": "b456ea0b-db46-42f4-9bbe-3df1a38e4fa8", "fetched_from_uri": "https://services2.arcgis.com/XZg2efAbaieYAXmu/ArcGIS/rest/services/Vaccine_Locations_nomad/FeatureServer", "fetched_at": "2021-06-16T23:01:59.865422", "published_at": "2021-01-25T10:01:37", "data": {"attributes": {"Activated1": "Yes", "Appointments": "No", "Contact": null, "CreationDate": 1611182038000, "Creator": "watsonem", "EditDate": 1611597697000, "Editor": "OyerZDHEC", "GlobalID": "b456ea0b-db46-42f4-9bbe-3df1a38e4fa8", "Loc_ID": 845, "OBJECTID": 1, "Phone_Sec": null, "Region": "Lowcountry", "SiteAddress": "137 gateway drive", "SiteAddressDetail": null, "SiteCity": "Ladson", "SitePhone": "", "SiteState": "SC", "SiteZip": "29456", "Status": "Approved", "URL": null, "V_Manufacturer": null, "layer_id": 0, "loc_admin_city": null, "loc_admin_phone": null, "loc_admin_state": null, "loc_admin_street1": null, "loc_admin_street2": null, "loc_admin_zip": null, "loc_covid_id": null, "loc_name": "Carolina Cataract and Vision Center", "loc_ship_city": "Ladson", "loc_ship_phone": "843-797-3676", "loc_ship_state": "SC", "loc_ship_street1": "137 gateway drive", "loc_ship_street2": null, "loc_ship_zip": 29456, "loc_type1": "Medical practice \u2013 other specialty", "ref_ID": 954, "service_item_id": "1256d5b3c554475099c24b98326e3b36", "vfc_text": "SCA918009", "vtrcks_ID": 40208317}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -80.09152585378803, "y": 32.997858505756824}}}}
{"id": "sc_nomad_arcgis:1256d5b3c554475099c24b98326e3b36_0_220d6067-6ceb-4655-93ef-b5ed05e84c49", "name": "Center for Primary Care, PC", "address": {"street1": "105 E Hugh St", "street2": null, "city": "North Augusta", "state": "SC", "zip": "29841"}, "location": {"latitude": 33.51771994180649, "longitude": -81.95652331218946}, "contact": null, "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": null, "active": false, "source": {"source": "sc_nomad_arcgis", "id": "220d6067-6ceb-4655-93ef-b5ed05e84c49", "fetched_from_uri": "https://services2.arcgis.com/XZg2efAbaieYAXmu/ArcGIS/rest/services/Vaccine_Locations_nomad/FeatureServer", "fetched_at": "2021-06-16T23:01:59.865422", "published_at": "2021-01-25T10:01:37", "data": {"attributes": {"Activated1": "No", "Appointments": "No", "Contact": null, "CreationDate": 1611182038000, "Creator": "watsonem", "EditDate": 1611597697000, "Editor": "OyerZDHEC", "GlobalID": "220d6067-6ceb-4655-93ef-b5ed05e84c49", "Loc_ID": 813, "OBJECTID": 2, "Phone_Sec": null, "Region": "Midlands", "SiteAddress": "105 E Hugh St", "SiteAddressDetail": null, "SiteCity": "North Augusta", "SitePhone": "", "SiteState": "SC", "SiteZip": "29841", "Status": "Approved", "URL": null, "V_Manufacturer": null, "layer_id": 0, "loc_admin_city": null, "loc_admin_phone": null, "loc_admin_state": null, "loc_admin_street1": null, "loc_admin_street2": null, "loc_admin_zip": null, "loc_covid_id": null, "loc_name": "Center for Primary Care, PC", "loc_ship_city": "North Augusta", "loc_ship_phone": "706-279-6800", "loc_ship_state": "SC", "loc_ship_street1": "105 E Hugh St", "loc_ship_street2": null, "loc_ship_zip": 29841, "loc_type1": "Medical practice \u2013 family medicine", "ref_ID": 955, "service_item_id": "1256d5b3c554475099c24b98326e3b36", "vfc_text": "SCA502005", "vtrcks_ID": 40117843}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -81.95652331218946, "y": 33.51771994180649}}}}
{"id": "sc_nomad_arcgis:1256d5b3c554475099c24b98326e3b36_0_6ef35196-c3f4-4464-a572-17f00f83c669", "name": "Seneca Health and Wellness Center/Cigna Onsite Health, LLC", "address": {"street1": "125 Lila Doyle Drive", "street2": null, "city": "Seneca", "state": "SC", "zip": "29678"}, "location": {"latitude": 34.69404697221887, "longitude": -82.98411190682685}, "contact": null, "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": null, "active": true, "source": {"source": "sc_nomad_arcgis", "id": "6ef35196-c3f4-4464-a572-17f00f83c669", "fetched_from_uri": "https://services2.arcgis.com/XZg2efAbaieYAXmu/ArcGIS/rest/services/Vaccine_Locations_nomad/FeatureServer", "fetched_at": "2021-06-16T23:01:59.865422", "published_at": "2021-01-25T10:01:37", "data": {"attributes": {"Activated1": "Yes", "Appointments": "No", "Contact": null, "CreationDate": 1611182038000, "Creator": "watsonem", "EditDate": 1611597697000, "Editor": "OyerZDHEC", "GlobalID": "6ef35196-c3f4-4464-a572-17f00f83c669", "Loc_ID": 782, "OBJECTID": 3, "Phone_Sec": null, "Region": "Upstate", "SiteAddress": "125 Lila Doyle Drive", "SiteAddressDetail": null, "SiteCity": "Seneca", "SitePhone": "", "SiteState": "SC", "SiteZip": "29678", "Status": "Approved", "URL": null, "V_Manufacturer": null, "layer_id": 0, "loc_admin_city": null, "loc_admin_phone": null, "loc_admin_state": null, "loc_admin_street1": null, "loc_admin_street2": null, "loc_admin_zip": null, "loc_covid_id": null, "loc_name": "Seneca Health and Wellness Center/Cigna Onsite Health, LLC", "loc_ship_city": "Seneca", "loc_ship_phone": "864-324-7969", "loc_ship_state": "SC", "loc_ship_street1": "125 Lila Doyle Drive", "loc_ship_street2": null, "loc_ship_zip": 29678, "loc_type1": "Health center \u2013 occupational", "ref_ID": 956, "service_item_id": "1256d5b3c554475099c24b98326e3b36", "vfc_text": "SCA937006", "vtrcks_ID": 40226365}, "geometry": {"spatialReference": {"latestWkid": 4326, "wkid": 4326}, "x": -82.98411190682685, "y": 34.69404697221887}}}}
```

## Before Opening a PR
- [X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by MoralCode** [comment]
yeah i noticed that too, i was ignoring it for the moment since it seems like its a bunch of location names shoved into one. 

I can update the scraper to just take the first one (that matches the addresS) if that helps, but it seems like the `Dorchester ( 4115 Dorchester Rd Charleston, SC 29405),` site is the only one that doesnt have its own separate record elsewhere and im not sure if/how its possible to split out multiple normalized records from a single record

File: vaccine_feed_ingest/runners/sc/dab_arcgis/normalize.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        if valid:
+            contacts.append(schema.Contact(contact_type="general", website=url))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+# Using "Appointments" field though unclear whether this should be interpreted as
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/sc/dab_arcgis/normalize.py:95]: Just remove this whole method it isn't doing anything useful

File: vaccine_feed_ingest/runners/sc/dhec_arcgis/normalize.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+# Using "Appointments" field though unclear whether this should be interpreted as
+# "An appointment is required" or "An appointment is available"
+def _get_activated(site: dict) -> bool:
+    if site["attributes"]["Activated1"] == "No":
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/sc/dhec_arcgis/normalize.py:98]: You can simplify this to `return site["attributes"]["Activated1"] != "No"`

**Review by bryanculbertson — COMMENTED** [review]
There is a lot of duplicate code between all of these. Could you move the methods into the shared utils?

**Comment by MoralCode** [comment]
I'm not sure if this code is universal enough to warrant being abstracted to that level. id prefer to combine the arcgis fetchers for the sources that are basically identical as far as the data keys go. does that sound preferable?

**Comment by bryanculbertson** [comment]
Okay, I'll just merge this then, and we can worry about DRYing it later

**Review by bryanculbertson — APPROVED** [review]

**Pull request merged**